File: python/29.py

all_powers = []
A, B = 100, 100
for a in range(2, A + 1):
    for b in range(2, B + 1):
        print("a: ", a, "b: ", b, "\r", end="")
        all_powers.append(a ** b)

print()
print(len(set(all_powers)))

File: cpp/29.cpp
#include <iostream>
#include <set>
#include <cmath>

using namespace std;


int main() {
    set<double> all_powers {};

    for (int a = 2; a < 101; a++)
    {
        for (int b = 2; b < 101; b++)
        {
            all_powers.insert(pow(a, b));
        }
        
    }

    cout << all_powers.size() << endl;
    return 0;
}